### PR TITLE
arm64, armv7, armv7s architecture targets

### PR DESCRIPTION
**Pull request by brunobowden**
Dependencies:
- j2objcTest dependsOn test (java plugin task)
- j2objcAssemble no longer dependsOn j2objcTest
- j2objcXcode dependsOn j2objcAssemble

Other:
- podspec debug library support
- j2objcHackToForceCompilation to better indicate purpose
- inline getInputStaticLibraryPath() as it's only used one place
- update list of main j2objc tasks

### CODE REVIEW

**Comment by brunobowden** [comment]
@advayDev1 @confile - please take a look

File: src/main/groovy/com/github/j2objccontrib/j2objcgradle/tasks/J2objcXcodeTask.groovy [hunk 1]
```diff
@@ -109,9 +109,9 @@ class J2objcXcodeTask extends DefaultTask {
                 "s.resources = '${j2objcResourceDirName}/**/*'\n" +
                 "s.requires_arc = true\n" +
                 "s.preserve_path = '${project.buildDir}/j2objc/*.a'\n" +
-                "s.libraries = 'ObjC', 'guava', 'javax_inject', 'jre_emul', 'jsr305', 'z', 'icucore', 'j2objc-generated-library'\n" +
+                "s.libraries = 'ObjC', 'guava', 'javax_inject', 'jre_emul', 'jsr305', 'z', 'icucore', '${project.name}-j2objc'\n" +
                 "s.xcconfig = { 'HEADER_SEARCH_PATHS' => '${getJ2ObjCHome()}/include', " +
-                "'LIBRARY_SEARCH_PATHS' => '${getJ2ObjCHome()}/lib ${project.buildDir}/j2objc' }\n" +
+                "'LIBRARY_SEARCH_PATHS' => '${getJ2ObjCHome()}/lib ${project.buildDir}/j2objcOutputs/lib/iosDebug' }\n" +
```
Comment (advayDev1) [inline src/main/groovy/com/github/j2objccontrib/j2objcgradle/tasks/J2objcXcodeTask.groovy]: comment on L101 can now be removed.

**Comment by advayDev1** [comment]
@brunobowden LGTM pending 1 minor comment

**Comment by advayDev1** [comment]
btw @confile to provide some context, @brunobowden and i discussed in person refactoring our dependencies to look more like java plugin.  namely if you ask for a j2objcAssemble (like java assemble) it will just create and copy the final binaries.  if you ask for a j2objcTest (like java test) it will just create and test the test binaries.  if you ask for a build, everything will occur.  no longer does every assembly require a test - so 'skipTest' becomes obsolete.

**Comment by advayDev1** [comment]
speaking of which, @brunobowden , perhaps you want to remove testSkip flag ?

**Comment by brunobowden** [comment]
removed comment
removed testSkip

other refactoring mentioned in comments and tested out

**Pull request merged**